src/syntax/syntax-check.rkt

```diff
@@ -38,6 +38,14 @@
     [#`(if #,args ...)
      (error! stx "Invalid `if` expression with ~a arguments (expects 3)" (length args))
      (unless (null? args) (check-expression* (last args) vars error!))]
+    [#`(! #,props ... #,body)
+     (let loop ([props props])
+       (match props
+         [(list) (void)]
+         [(list prop) (error! stx "Expected a value for property ~a" prop)]
+         [(list (app syntax-e (? symbol? prop)) value rest ...) (loop rest)]
```

Comment:
Instead of `app syntax-e` you can just use `#,`. Also, we don't just want it to be a symbol, right? We want it to start with a colon too?